### DISCUSSION TITLE
Any idea how do I fetch external API and display it within the Strapi dashboard?

### DISCUSSION THREAD

**Discussion by dunods** [strapi/strapi#5925] [Legacy Topics]
hello,

is anyone here have experience to fetch data from external api and display it within strapi admin?
and using all the functionality of content-manager to manage it (crud function)?
for example given this extenal api as an example 
https://jsonplaceholder.typicode.com/posts

please assist

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by lauriejim** [comment]
Hello! Do you have the access of the database?
To create/update/delete data? Not only fetch.

Or you want to map the Strapi admin on an API instead of a database?
Not sure to exactly understand what you are trying to do.

**Reply by dunods** [reply]
well the use case are assuming that we are given external rest api that enable all create, update, and delete.  and want to be able to access and edit that data respectively within strapi admin .
any thought ?

**Reply by soupette** [reply]
You can create a view in the content manager so here are some thoughts:

1. First create a new `<Route />` by extending this [file]( https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-content-manager/admin/src/containers/Main/index.js)
2. Then if you want the link to appear in the left menu you should look at [this one](https://github.com/strapi/strapi/blob/master/packages/strapi-plugin-content-manager/admin/src/containers/Initializer/index.js) in order to inject a link in either the collection type or content type section

**Reply by dunods** [reply]
i dont think we can achieve that by doing so.

to make it simple, we are given this api 
[GET ALL] localhost:3000/api/v1/product/ 
[GET BY ID] localhost:3000/api/v1/product/:id
[POST] localhost:3000/api/v1/product
[PATCH]localhost:3000/api/v1/product/:id
[DELETE] localhost:3000/api/v1/product/:id

any clue/help ??

**Comment by kyuumeitai** [comment]
hey @dunods maybe you can do a lambda/openfaas function that writes that data into Strapi, no?

**Reply by dunods** [reply]
thats not what im trying to achieve, where the scenario are the strapi has its own database as a micro services and there are some other external api that we want to manage/manipulate within the strapi admin dashboard. yes we can trigger callback by creating lambda function to create object to strapi, but by doing so we will have lotsa data redundancy.

**Comment by codeplott** [comment]
Hey,
have you found a solution?

**Reply by Josh-Uvi** [reply]
@dunods @sawden hope it's not too late, this is the solution I found https://strapi.io/documentation/v3.x/guides/external-data.html#content-type-settings

**Reply by lauriejim** [reply]
And if you want to display stuff in your admin, you will have to customize your admin by creating a plugin (to have a new custom interface)

**Comment by derrickmehaffy** [comment]
Locking as migrated to: https://forum.strapi.io/t/any-idea-how-do-i-fetch-external-api-and-display-it-within-the-strapi-dashboard-5925/1493
